pkgs/by-name/cq/cq/package.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, fetchFromGitHub
+, fetchurl
+, buildGraalvmNativeImage
+, graalvmCEPackages
+}:
+
+buildGraalvmNativeImage rec {
+  pname = "cq";
+  version = "2024.02.13-23.14";
+
+  # we need both src (the prebuild jar)
+  src = fetchurl {
+    url = "https://github.com/markus-wa/cq/releases/download/${version}/cq.jar";
+    sha256 = "sha256-JQp9fBR7jhenXcXSVMFsWosKUsjNxWVU+QnTz0X+dwk=";
+  };
+
+  # and build-src (for the native-image build process)
+  build-src = fetchFromGitHub {
+    owner = "markus-wa";
+    repo = "cq";
+    rev = version;
+    hash = "sha256-Ir4IrUucouKz3gPPrpCigyb+utE8t0I1Hzb/Rb88haM=";
+  };
+
+  graalvmDrv = graalvmCEPackages.graalvm-ce;
+
+  executable = "cq";
+
+  # copied verbatim from the upstream build script https://github.com/markus-wa/cq/blob/main/package/build-native.sh#L5
+  extraNativeImageBuildArgs = [
+    "--report-unsupported-elements-at-runtime"
+    "--initialize-at-build-time"
+    "--no-server"
+    "-H:ReflectionConfigurationFiles=${build-src}/package/reflection-config.json"
+  ];
+
+  meta = with lib; {
+    description = "Clojure Query: A Command-line Data Processor for JSON, YAML, EDN, XML and more";
+    homepage = "https://github.com/markus-wa/cq";
```

Comment:
For better package metadata you could add,

```
    changelog = "https://github.com/markus-wa/cq/releases/releases/tag/${version}";
```

and this will also trigger the update bot to automatically submit PRs for new versions.

Reply:
TIL, thanks